### PR TITLE
first shot at the integration testing environment

### PR DESCRIPTION
**Pull request by jbendsen**
I've made a new docker-compose file which only exposes port 8080 externally to minimize port  conflicts. Also I've described how to build and test locally in a README.md file. Let me know what you think.

### CODE REVIEW

File: demo-project/demo-project-integration-test/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+# How to run
+## 1. Create docker image
+First you have to create a Docker image for the Demo Spring Boot application.
+This is done by issuing a Maven command:
+```sh
+cd demo-project #make sure to cd to the .. folder
+mvn spring-boot:build-image -Dspring-boot.build-image.imageName=logb/demo-project
+```
+This will build a Docker image with the name logb/demo-project:latest.
```
Comment (jypma) [inline demo-project/demo-project-integration-test/README.md]: We should investigate versioning a little. Like, doesn't the maven version get applied as the docker tag as well? In general, working with `latest` on anything else than locally has a high risk of grabbing the wrong version.

File: demo-project/demo-project-integration-test/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+The [docker-compose.yml](./docker-compose.yml]) file defines 3 containers and you can start the containers like this:
+
+```sh
+cd demo-project/demo-project-integration-test
+docker-compose up -d 
+```
+
+You can verify that the containers are running by making a GET request to this URL:
+```
+http://localhost:8080/values
```
Comment (jypma) [inline demo-project/demo-project-integration-test/README.md]: may as well include `sh` here and `curl`, to make it runnable.

File: demo-project/demo-project-integration-test/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  db:
+    image: postgres:13.2-alpine
+    # Uncomment this to have the DB come up when you start docker / your laptop:
+    #restart: always
+    environment:
+      POSTGRES_USER: demo
+      POSTGRES_DB: demo
+      POSTGRES_PASSWORD: example
+    ports:
+      - 5432
```
Comment (jypma) [inline demo-project/demo-project-integration-test/docker-compose.yml]: don't bind ports to the host for integration tests

File: demo-project/demo-project-integration-test/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+  rabbitmq:
+    image: rabbitmq:3.8.16-alpine
+    # Uncomment this to have the DB come up when you start docker / your laptop:
+    #restart: always
+    ports:
```
Comment (jypma) [inline demo-project/demo-project-integration-test/docker-compose.yml]: don't bind ports to the host for integration tests

File: demo-project/demo-project-integration-test/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+      - 8080:8080
+    depends_on:
+      - db
+      - rabbitmq
+
+  db:
+    image: postgres:13.2-alpine
+    # Uncomment this to have the DB come up when you start docker / your laptop:
+    #restart: always
```
Comment (jypma) [inline demo-project/demo-project-integration-test/docker-compose.yml:21]: remove these comments, integration tests should never restart

**Review by jypma — CHANGES_REQUESTED** [review]

File: demo-project/demo-project-integration-test/README.md (hunk 1 above)
Comment (jbendsen) [reply demo-project/demo-project-integration-test/README.md]: This works: mvn spring-boot:build-image -Dspring-boot.build-image.imageName=logb/demo-project:**${project.version}**. However we still need to update the reference to the image in the docker-compose.yml file. Is this acceptable?

File: demo-project/demo-project-integration-test/README.md (hunk 2 above)
Comment (jbendsen) [reply demo-project/demo-project-integration-test/README.md]: done

File: demo-project/demo-project-integration-test/docker-compose.yml (hunk 4 above)
Comment (jbendsen) [reply demo-project/demo-project-integration-test/docker-compose.yml]: done

File: demo-project/demo-project-integration-test/docker-compose.yml (hunk 5 above)
Comment (jbendsen) [reply demo-project/demo-project-integration-test/docker-compose.yml:21]: done

**Comment by jbendsen** [comment]
@jypma, I've pushed some changes that hopefully resolves your comments.

File: demo-project/demo-project-integration-test/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  app:
+    image: logb/demo-project:0.0.1-SNAPSHOT
+    environment:
+      SPRING_DATASOURCE_URL: jdbc:postgresql://db:5432/demo
+      SPRING_DATASOURCE_USERNAME: demo
+      SPRING_DATASOURCE_PASSWORD: example
+      SPRING_RABBITMQ_HOST: rabbitmq
+      SPRING_RABBITMQ_PORT: 5672
+    ports:
+      - 8080:8080
```
Comment (jypma) [inline demo-project/demo-project-integration-test/docker-compose.yml:13]: Let's pick a different port here, so you can have the local one simultaneoulsy

File: demo-project/demo-project-integration-test/README.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+The [docker-compose.yml](./docker-compose.yml]) file defines 3 Docker containers: Spring Boot, Postgresql, and RabbitMq. 
+
+# How to run
+## 1. Create docker image
+First you have to create a Docker image for the Demo Spring Boot application.
+This is done by issuing a Maven command:
+```sh
+cd demo-project #make sure to cd to the .. folder
+mvn spring-boot:build-image -Dspring-boot.build-image.imageName=logb/demo-project:${project.version}
```
Comment (jypma) [inline demo-project/demo-project-integration-test/README.md:13]: escape the `\$` (or make two alternatives, windows and linux)

File: demo-project/demo-project-integration-test/docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,29 @@
+version: '3.1'
+
+services:
+  app:
+    image: logb/demo-project:0.0.1-SNAPSHOT
```
Comment (jypma) [inline demo-project/demo-project-integration-test/docker-compose.yml:5]: Good enough to just say `integration-test` here as a fixed string tag for the docker image.

**Pull request merged**